File: ClientApp/src/app/model/production-dtos.ts
// Production Request DTOs
export interface ProducedProductRequest {
  id?: number;
  configuredProductId: number;
  bestandteilRequests: BestandteilRequest[];
}

export interface BestandteilRequest {
  globalAssetId: string;
  usageDate: Date;
  amount: number;
}

// Production Response DTOs
export interface ProductionResponse {
  success: boolean;
  message: string;
  producedProduct?: import('./produced-product').ProducedProduct;
  error?: string;
}

// HandoverDocumentation DTOs
export interface HandoverDocumentationResponse {
  success: boolean;
  message: string;
  submodel?: SubmodelSummary;
  error?: string;
}

export interface SubmodelSummary {
  id: string;
  idShort?: string;
  description?: string;
  version?: string;
  revision?: string;
  elementCount: number;
}

File: Database/Model/DTOs/ProductionDtos.cs
using System;
using System.Collections.Generic;

namespace AasDemoapp.Database.Model.DTOs
{
    public class ProducedProductRequestDto
    {
        public long? Id { get; set; }
        public long ConfiguredProductId { get; set; }
        public List<BestandteilRequestDto> BestandteilRequests { get; set; } = [];
    }

    public class BestandteilRequestDto
    {
        public string GlobalAssetId { get; set; } = string.Empty;
        public DateTime UsageDate { get; set; }
        public int Amount { get; set; } = 1;
    }

    public class ProductionResponseDto
    {
        public bool Success { get; set; }
        public string Message { get; set; } = string.Empty;
        public ProducedProductDto? ProducedProduct { get; set; }
        public string? Error { get; set; }
    }
}
